src/application/services/curx-address-resolver.ts

```diff
@@ -0,0 +1,78 @@
+import { CruxUser, IAddress } from "../../core/entities/crux-user";
+import { BaseError, ErrorHelper, getLogger, PackageErrorCode } from "../../packages";
+import { CruxAssetTranslator, IAssetMatcher } from "./crux-asset-translator";
+const log = getLogger(__filename);
+export interface ICruxAddressResolverOptions {
+    cruxAssetTranslator: CruxAssetTranslator;
+    cruxUser: CruxUser;
+    userCruxAssetTranslator?: CruxAssetTranslator;
+}
+export class CruxAddressResolver {
```

Comment:
We shouldn't need this component at application service level. 
Apart from translation of symbol, we need to move everything inside the user.

The AssetTranslator class also should not need any changes. We are not making any changes to translation process.